packages/server/src/core/internals/procedureBuilder.ts

```diff
@@ -42,6 +43,19 @@ type CreateProcedureReturnInput<
   _output_out: FallbackValue<TNext['_output_out'], TPrev['_output_out']>;
 }>;
 
+type CreateProcedureReturnInput2<
```

Comment:
This type needs a lot of work, it's the main source of bugs and is clearly not named well, can be largely ignored for now :)